frontend/src/api/trustyai/custom.ts

```diff
@@ -18,17 +19,19 @@ export const getSpdRequests =
 export const createSpdRequest =
   (hostPath: string) =>
   (opts: K8sAPIOptions, data: BaseMetricRequest): Promise<BaseMetricCreationResponse> =>
-    proxyCREATE(hostPath, '/metrics/spd/request', data, {}, opts);
+    handleTrustyAIFailures(proxyCREATE(hostPath, '/metrics/spd/request', data, {}, opts));
 
 export const deleteSpdRequest =
   (hostPath: string) =>
   (opts: K8sAPIOptions, id: string): Promise<void> =>
-    proxyDELETE(
-      hostPath,
-      '/metrics/spd/request',
-      { requestId: id },
-      {},
-      { parseJSON: false, ...opts },
+    handleTrustyAIFailures(
+      proxyDELETE(
+        hostPath,
+        '/metrics/spd/request',
+        { requestId: id },
+        {},
+        { parseJSON: false, ...opts },
+      ),
     );
 
 export const getDirRequests =
```

Comment:
Do we not need to handle the error for get requests also?

Reply:
Hmm, I haven't found any get errors yet, but we can definitely handle them all.